students/HarveyHerela/lesson09/mailroom_actions.py

```diff
@@ -0,0 +1,68 @@
+def send_thank_you(db, firstname, lastname, amount):
+    db.add_donation(firstname, lastname, amount)
+    return "Thank you {fn} {ln} for your generous donation of ${amount:.2f}".format(
+        fn=firstname, ln=lastname, amount=amount)
+
+
+def create_report(db):
+    """Prints a list of all donors, sorted by total amount given."""
+
+    def get_total(donor_data):
+        return donor_data[1]
+
+
+    def create_row(donor):
+        fullname = donor.get_name()
+        total = sum(donor.get_donations())
+        num = len(donor.get_donations())
+        avg = total / num
+        return [fullname, total, num, avg]
```

Comment:
You could store these calculated donation values as `@properties` on the `Donor` object